File: src/api/test_upload.py
import pytest
from fastapi.testclient import TestClient
from src.api.upload import app

client = TestClient(app)

def test_upload_file():
    with open("test_file.txt", "w") as f:
        f.write("This is a test file.")
    with open("test_file.txt", "rb") as f:
        response = client.post("/upload", files={"upload_file": f})
        assert response.status_code == 200
        assert response.json() == {"filename": "test_file.txt"}

File: src/api/test_upload.js
const request = require('supertest');
const app = require('../server'); // Import the server setup

describe('POST /api/upload', () => {
    it('should upload a video file successfully', async () => {
        const response = await request(app)
            .post('/api/upload')
            .attach('video', 'test/video.mp4'); // Make sure to have a test video in your test folder
        expect(response.status).toBe(200);
        expect(response.body.message).toBe('Video uploaded successfully!');
    });

    it('should return error for invalid file type', async () => {
        const response = await request(app)
            .post('/api/upload')
            .attach('video', 'test/image.png'); // Testing with an invalid file type
        expect(response.status).toBe(400);
        expect(response.body.message).toBe('Error: Video files only!');
    });
});